src/Akka.Streams.Kafka.Tests/BugFix240SupervisionStrategy.cs

```diff
@@ -197,70 +197,66 @@ public async Task Committable_consumer_with_failed_downstream_stage_result_shoul
                 .WithStopTimeout(TimeSpan.FromSeconds(1))
                 .WithProperty("auto.offset.reset", "earliest")
                 .WithGroupId(group);
-
-            var counter = 0;
             
+            var committerSettings = CommitterSettings.Create(Sys);
+
             await Source.From(Enumerable.Range(1, 11))
                 .Select(elem => new ProducerRecord<Null, string>(topicPartition, elem.ToString()))
                 .RunWith(KafkaProducer.PlainSink(ProducerSettings), Materializer);
-            
-            var probe = KafkaConsumer.CommittableSource(consumerSettings, Subscriptions.AssignmentWithOffset(new TopicPartitionOffset(topicPartition, Offset.Unset)))
-                .Select(t =>
+
+            var count = 0;
+            var (control, probe) = KafkaConsumer.CommittableSource(consumerSettings, Subscriptions.AssignmentWithOffset(new TopicPartitionOffset(topicPartition, Offset.Beginning)))
+                .Select(c =>
                 {
-                    counter++;
-                    // fail once, on the 7th message
-                    if (counter == 7)
+                    if(++count == 7)
                         throw new Exception("BOOM!");
-                    return t;
-                })
-                .SelectAsync(1, async elem =>
-                {
-#pragma warning disable CS0618 // Type or member is obsolete
-                    await elem.CommitableOffset.Commit();
-#pragma warning restore CS0618 // Type or member is obsolete
-                    return elem.Record.Message.Value;
+                    return c;
                 })
-                .ToMaterialized(this.SinkProbe<string>(), Keep.Right)
+                .AlsoToMaterialized(CreateSink(), Keep.Left)
+                .Select(c => c.Record.Message.Value)
+                .ToMaterialized(this.SinkProbe<string>(), Keep.Both)
                 .Run(Materializer);
 
-            var messages = new List<string>();
-            probe.Request(11);
-            for (var i = 0; i < 6; i++)
-            {
-                messages.Add(probe.ExpectNext(TimeSpan.FromSeconds(5))); 
-            }
+            var offsets = new List<string>();
+            await probe.RequestAsync(11);
+            offsets.AddRange(await probe.ExpectNextNAsync(6).ToListAsync()); // we get an extra element here even though the offset is not committed
```

Comment:
So this was super-frustrating to debug, but after I got the `CommitCollectorStage` in-place with its logging I could see what was going on: we were emitting 1 more element than what we'd actually committed. This test worked before, ironically, by _missing an offset_. I fixed that as part of this PR.